pandas/core/generic.py

```diff
@@ -6153,7 +6154,7 @@ def __finalize__(self, other, method: str | None = None, **kwargs) -> Self:
         """
         if isinstance(other, NDFrame):
             for name in other.attrs:
-                self.attrs[name] = other.attrs[name]
+                self.attrs[name] = deepcopy(other.attrs[name])
```

Comment:
Do we want to deepcopy unconditionally. `copy.deepcopy(df)` goes to `df.copy(deep=True)` (`deep=True` also being the default). By this change, `df.copy(deep=False)` would also deepcopy. If we don't want this, we'd have to pass the copy mode from `df.copy()` down to `__finalize__`. - One way to do this is by passing a specialized *method* parameter.